Code/GraphMol/Substruct/SubstructUtils.cpp

```diff
@@ -103,6 +103,25 @@ class ScoreMatchesByDegreeOfCoreSubstitution {
 };
 }  // namespace detail
 
+bool propertyCompat(const RDProps *r1, const RDProps *r2,
+                    const std::vector<std::string>& properties) {
+  std::string prop1;
+  std::string prop2;
+  for (const auto &prop : properties) {
+    bool hasprop1 = r1->getPropIfPresent<std::string>(prop, prop1);
+    bool hasprop2 = r2->getPropIfPresent<std::string>(prop, prop2);
```

Comment:
```suggestion
  PRECONDITION(r1,"bad RDProps");
  PRECONDITION(r2,"bad RDProps");
  
  for (const auto &prop : properties) {
    std::string prop1;
    bool hasprop1 = r1->getPropIfPresent<std::string>(prop, prop1);
    std::string prop2;
    bool hasprop2 = r2->getPropIfPresent<std::string>(prop, prop2);
```

Reply:
Doesn't look like this change has been made. I really think it's a good idea to have the `PRECONDITION` calls whenever a function takes a naked pointer and dereferences it without checking

Reply:
Oh thanks, I somehow missed this originally!